src/utils/fetchClient.ts

```diff
@@ -29,7 +29,10 @@ function request<T>(
   // for a demo purpose we emulate a delay to see if Loaders work
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+    .then(response => response.json())
+    .catch(() => {
+      throw new Error();
```

Comment:
The catch block throws a generic error without any message. Consider providing a more descriptive error message to aid in debugging.